core/templates/dev/head/domain/skill/SkillObjectFactorySpec.ts

```diff
@@ -15,234 +15,205 @@
-// ^^^ This block is to be removed.
-
-require('App.ts');
-require('domain/skill/ConceptCardObjectFactory.ts');
-require('domain/skill/SkillObjectFactory.ts');
-
-describe('Skill object factory', function() {
-  beforeEach(angular.mock.module('oppia'));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    $provide.value(
-      'AudioLanguageObjectFactory', new AudioLanguageObjectFactory());
-    $provide.value(
-      'AutogeneratedAudioLanguageObjectFactory',
-      new AutogeneratedAudioLanguageObjectFactory());
-    $provide.value(
-      'MisconceptionObjectFactory', new MisconceptionObjectFactory());
-    $provide.value(
-      'RubricObjectFactory', new RubricObjectFactory());
-    $provide.value(
-      'RecordedVoiceoversObjectFactory',
-      new RecordedVoiceoversObjectFactory(new VoiceoverObjectFactory()));
-    $provide.value(
-      'SubtitledHtmlObjectFactory', new SubtitledHtmlObjectFactory());
-    $provide.value('VoiceoverObjectFactory', new VoiceoverObjectFactory());
-  }));
-  beforeEach(angular.mock.module('oppia', function($provide) {
-    var ugs = new UpgradedServices();
-    for (let [key, value] of Object.entries(ugs.getUpgradedServices())) {
-      $provide.value(key, value);
-    }
-  }));
-
-  describe('SkillObjectFactory', function() {
-    var SkillObjectFactory = null;
-    var MisconceptionObjectFactory = null;
-    var SubtitledHtmlObjectFactory = null;
-    var RubricObjectFactory = null;
-    var ConceptCardObjectFactory = null;
-    var misconceptionDict1 = null;
-    var misconceptionDict2 = null;
-    var rubricDict = null;
-    var skillContentsDict = null;
-    var skillDict = null;
-    var skillDifficulties = null;
-
-    beforeEach(angular.mock.inject(function($injector) {
-      SkillObjectFactory = $injector.get('SkillObjectFactory');
-      MisconceptionObjectFactory = $injector.get('MisconceptionObjectFactory');
-      RubricObjectFactory = $injector.get('RubricObjectFactory');
-      ConceptCardObjectFactory = $injector.get('ConceptCardObjectFactory');
-      SubtitledHtmlObjectFactory = $injector.get('SubtitledHtmlObjectFactory');
-      skillDifficulties = $injector.get('SKILL_DIFFICULTIES');
-
-      misconceptionDict1 = {
-        id: 2,
-        name: 'test name',
-        notes: 'test notes',
-        feedback: 'test feedback',
-        must_be_addressed: true
-      };
-
-      misconceptionDict2 = {
-        id: 4,
-        name: 'test name',
-        notes: 'test notes',
-        feedback: 'test feedback',
-        must_be_addressed: false
-      };
-
-      rubricDict = {
-        difficulty: skillDifficulties[0],
-        explanation: 'explanation'
-      };
-
-      skillContentsDict = {
-        explanation: {
-          html: 'test explanation',
-          content_id: 'explanation',
+import { TestBed } from '@angular/core/testing';
+const constants = require('constants.ts');
+
+describe('Skill object factory', () => {
+  let skillObjectFactory;
```

Comment:
All the variables should be assigned first as `null`.

Reply:
changed all the var to let